### PR TITLE
Ttgo tests

### PR DESCRIPTION
**Pull request by SlupikB**

### CODE REVIEW

File: code/ttgo_tests/Arduino_Micro_Sender/Arduino_Micro_Sender.ino [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    Serial.println("LoRa Initialization failed:");  
+    while(1);
+  }
+  Serial.println("LoRa initialization successful");
+}
+
+void loop() {
+  // put your main code here, to run repeatedly:
+  digitalWrite(13, HIGH);
+  Serial.print("Sending packet: ");
```
Comment (evemorgen) [inline code/ttgo_tests/Arduino_Micro_Sender/Arduino_Micro_Sender.ino]: string concatenation might be more readable. For ex.
```
Serial.println("Sending packet: " + counter);
```

File: code/ttgo_tests/Arduino_Micro_Sender/Arduino_Micro_Sender.ino [hunk 2]
```diff
@@ -0,0 +1,35 @@
+#include <LoRa.h> 
+int counter = 0;
+
+void setup() {
+  // put your setup code here, to run once:
+  pinMode(13, OUTPUT);
+  digitalWrite(13, LOW);
+  Serial.begin(9600);
+  /*while (!Serial){
```
Comment (evemorgen) [inline code/ttgo_tests/Arduino_Micro_Sender/Arduino_Micro_Sender.ino]: I assume that this comment is unnecessary? :)

File: code/ttgo_tests/Arduino_Micro_Sender_Receiver/Arduino_Micro_Sender_Receiver.ino [hunk 3]
```diff
@@ -0,0 +1,96 @@
+#include <LoRa.h> //External library by Sandeep Mistry https://github.com/sandeepmistry/arduino-LoRa
+
+int counter = 0; //Counting packets
+
+String packSize ;
```
Comment (evemorgen) [inline code/ttgo_tests/Arduino_Micro_Sender_Receiver/Arduino_Micro_Sender_Receiver.ino]: what happend to these semicolons? 😛

File: code/ttgo_tests/TTGO_Sender_Receiver/TTGO_Sender_Receiver.ino [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  display.drawStringMaxWidth(0 , 15 , 128, "RX:  " + packet);
+  display.drawString(0, 0, "RSSI: "+ rssi);  
+  display.display();
+}
+
+void readPacket(int packetSize) {
+  //Reads packet from radio socket
+  packet ="";
+  packSize = String(packetSize,DEC);
+  for (int i = 0; i < packetSize; i++) {
```
Comment (evemorgen) [inline code/ttgo_tests/TTGO_Sender_Receiver/TTGO_Sender_Receiver.ino]: reading by one byte makes no sense to me when `LoRa` object provides [`steams` capability](https://www.arduino.cc/reference/en/language/functions/communication/stream/)

File: code/ttgo_tests/Arduino_Micro_Sender_Receiver/Arduino_Micro_Sender_Receiver.ino [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  }
+  Serial.print("Received Packet: ");
+  Serial.println(packet);
+  packetsRcvd++;
+}
+
+void sendPacket(){
+   //Transmits packet with following properties
+   //Returns number sent by TTGO
+   sentPacket = "";
```
Comment (evemorgen) [inline code/ttgo_tests/Arduino_Micro_Sender_Receiver/Arduino_Micro_Sender_Receiver.ino]: why is this variable global?

File: code/ttgo_tests/TTGO_Sender_Receiver/TTGO_Sender_Receiver.ino (hunk 4 above)
Comment (SlupikB) [reply code/ttgo_tests/TTGO_Sender_Receiver/TTGO_Sender_Receiver.ino]: I have changed this to LoRa.readString() (a Stream property which reads streamed data into a string, compiles correctly), have no idea if it works though as i'm currently not in possession of the test board.

File: code/ttgo_tests/Arduino_Micro_Sender_Receiver/Arduino_Micro_Sender_Receiver.ino [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+   LoRa.beginPacket();
+   LoRa.print(sentPacket);
+   LoRa.endPacket();
+   packetsSent++;
+}
+
+void setup() {
+  // put your setup code here, to run once:
+  pinMode(13, OUTPUT);
```
[the remark this answers is not in the document]
Comment (evemorgen) [reply code/ttgo_tests/Arduino_Micro_Sender_Receiver/Arduino_Micro_Sender_Receiver.ino]: It is not about simplifying three lines, but to apply some mindset. Imagine that you've got code like this:
```
void setup() {
  pinMode(13, OUTPUT);
  pinMode(12, INPUT);
  pinMode(11, OUTPUT);
}
```
How would you know what is the purpose of these `pinModes`? Maybe pin 13 were set to output for being LED driver, or maybe for SPI interface?
Now, consider this one
```
void setup() {
  led = LED(10);
  spi = SPI(13,12,11); 
}
```
Readability and easier reasoning definitely counts :)

**Comment by evemorgen** [comment]
> 
> 
> @jjp241 Sholudn't loraTestsKuba directory be in /code/ttgo_tests rather than /?

It needs to be in different PR

File: code/ttgo_tests/TTGO_Sender_Receiver/TTGO_Sender_Receiver.ino (hunk 4 above)
Comment (evemorgen) [reply code/ttgo_tests/TTGO_Sender_Receiver/TTGO_Sender_Receiver.ino]: I guess that will do

**Comment by evemorgen** [comment]
[First sensors readings](https://trello.com/c/I53laWAU/57-first-sensors-readings)

**Pull request merged**